components/eamxx/src/physics/rrtmgp/CMakeLists.txt

```diff
@@ -2,40 +2,41 @@ include(EkatUtils)
 include(EkatSetCompilerFlags)
 include(ScreamUtils)
 
+##################################
+#             YAKL               #
+##################################
+
+# In case other E3SM components are building YAKL...
+if (TARGET yakl)
+  string (CONCAT msg
+    "It appears other parts of E3SM already built YAKL.\n"
+    "We are not preopared for this. Please, check the cmake\n"
+    "logic in EAMxx, so that we can use an existing yakl.")
+  message("${msg}")
+  message(FATAL_ERROR "Error! Aborting...")
+endif()
+# RRTMGP++ requires YAKL
+
 # Add CUDA flags for YAKL
-set(YAKL_CXX_FLAGS "")
 if (CUDA_BUILD)
   set(YAKL_ARCH "CUDA")
-  list(APPEND YAKL_CXX_FLAGS -DYAKL_ARCH_CUDA --expt-extended-lambda --expt-relaxed-constexpr)
+  set(YAKL_CUDA_FLAGS "-DYAKL_ARCH_CUDA --expt-extended-lambda --expt-relaxed-constexpr -ccbin ${CMAKE_CXX_COMPILER}")
 endif()
 if (HIP_BUILD)
   set(YAKL_ARCH "HIP")
-  list(APPEND YAKL_CXX_FLAGS "-DYAKL_ARCH_HIP -O3 -D__HIP_ROCclr__ -D__HIP_ARCH_GFX90A__=1 --rocm-path=${ROCM_PATH} --offload-arch=gfx90a -x hip ")
+  set(YAKL_HIP_FLAGS "-DYAKL_ARCH_HIP -O3 -D__HIP_ROCclr__ -D__HIP_ARCH_GFX90A__=1 --rocm-path=${ROCM_PATH} --offload-arch=gfx90a -x hip")
 endif()
 
+set (YAKL_SOURCE_DIR ${SCREAM_BASE_DIR}/../../externals/YAKL)
+add_subdirectory(${YAKL_SOURCE_DIR} ${CMAKE_BINARY_DIR}/externals/YAKL)
```

Comment:
This might conflict with how YAKL is built in `build_model.cmake` for CIME-based builds. Have you tried a CIME-based case?

Reply:
No, but this is exactly how it was built before (I just put the path in a var, to make it clearer). Why would it conflict though?

Reply:
I see what you mean, and now I'm wondering how it managed to work so far...

Reply:
Ah wait, that usage of YAKL is inside `if (COMP_NAME STREQUAL "eam")`, so it won't happen if we build eamxx. Still, I am adding a guard in case other components (e.g., Omega ocean) are building YAKL. Later, we might have to add a check to see if YAKL was pre-built as part of the sharedlib phase...